### DISCUSSION TITLE
Back behavior

### DISCUSSION THREAD

**Discussion by BraisGabin** [community/community#7625] [Mobile]
Right now when I open a PR from the notifications the app only shows me that someone approved it and when I go back I go back to the notifications. That's useless because I can do nothing regarding that pr! If I get an approve I probably want to merge it. Or at least see the general status to decide.

Probably the button up from the screen should make me back to the pr meanwhile the button back should just make me back to notifications but we need a way to go to the pr.

**Comment by brianlovin** [comment]
Thanks for the heads up @BraisGabin — the team is discussing the best solution for this right now, and we'll get a fix out as soon as possible to make this experience better.

**Reply by BraisGabin** [reply]
Same issue with the notification related to this comment. No way I can re read my original request.

**Reply by IWareQ** [reply]
> Same issue with the notification related to this comment. No way I can re read my original request. 

+1

**Comment by ibuclaw** [comment]
It's gotten marginally worse since the last couple updates too.  Now the back button takes you to the repository's home page instead of returning you to the notifications tab.

**Comment by brianlovin** [comment]
We're working on fixes for these issues now, thanks for the reports everyone!

**Comment by BudgieInWA** [comment]
On Android, the "up vs. back" distinction needs to be respected, which would solve a whole suite of problems like this (as well as actually making it possible to understand how navigation works before using the app for 100 hours) (see [related](https://github.com/github-community/community/discussions/3916)). I'm not as sure on iOS, but I thought the in-app up button is also supposed to be "up" and not "back".

My initial thought would be to implement the in-app "up" button as strictly "up" in the hierarchy, but allow items to exist in multiple places in the hierarchy. E.g. an issue could exist at *Home > Repo > Issue* and *Notifications > Issue*, the applicable hierarchy depending on how the user navigated there.
